pkg/build/builder/common.go

```diff
@@ -180,3 +155,47 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateBuildRevision(build *api.Build, sourceInfo *git.SourceInfo) *api.SourceRevision {
+	if build.Spec.Revision != nil {
+		return build.Spec.Revision
+	}
+	return &api.SourceRevision{
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+}
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface, sourceRev *api.SourceRevision) error {
+	return kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
```

Comment:
I spoke with @mfojtik who told me that @kargakis says we don't use patch from within controllers.  Can someone elaborate on why/why not and clarify our philosophy here?

Certainly we have plenty of other places where we do the pattern shown in the current code.